### PR TITLE
fix(client): add superBlock check to handle Rosetta being treated as a project block

### LINKED ISSUE #46006: Missing course collapse/expand button for Rosetta Code
### Describe the Issue

Rosetta Code is missing expand/collapse button. 

Issue is caused as a combination of https://github.com/freeCodeCamp/freeCodeCamp/blob/64cb47e1348d63e38a60d78646c1234f1257ecfd/client/src/templates/Introduction/components/block.tsx#L124-L126 with rosetta code having order `3` and challengeType `5`. As such it is treated as projects block.

One possible way of dealing is changing the check above, to include superBlock check. Another might be using different order in meta for Rosetta.

### Affected Page

https://www.freecodecamp.org/learn/coding-interview-prep/

### Steps to Reproduce

1. Go to https://www.freecodecamp.org/learn/coding-interview-prep/
2. Scroll to Rosetta Code

### Expected behavior

Have expand/collapse button for Rosetta Code

### Screenshots

image

### System

N/A

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by JordanMooree**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #46006

The code adds the following condition:
`challenge.superBlock !== 'coding-interview-prep'` as the superBlock check.

It produces the following result for Rosetta:
image

Hope it resolved the issue!

### CODE REVIEW

**Review by gikf — DISMISSED** [review]
Looks good :+1:

File: client/src/templates/Introduction/components/block.tsx [hunk 1]
```diff
@@ -122,7 +122,8 @@ export class Block extends Component<BlockProps> {
     const isProjectBlock = challenges.some(({ challenge }) => {
       const isJsProject =
         [3, 6, 10, 14, 17].includes(challenge.order) &&
-        challenge.challengeType === 5;
+        challenge.challengeType === 5 &&
+        challenge.superBlock !== 'coding-interview-prep';
```
Comment (raisedadead) [inline client/src/templates/Introduction/components/block.tsx:126]: 🙄

**Review by raisedadead — COMMENTED** [review]
@JordanMooree 

Thanks for the PR and your hard work. 

Everything below this line is not for you directly. It's for us, maintainers.

--

This change should fix the symptom, but trust me, it stinks, to be blunt. Can we make this better @freeCodeCamp/dev-team 

I don't particularly appreciate that we need to use the `isJsProject` as a classifier for this proposed fix.

Also, the whole logic in this component is a bit convoluted.

**Comment by ojeytonwilliams** [comment]
We've mixed up concerns. The curriculum should be telling the client what to display and the client should just render it, rather than having to first figure out what it's supposed to be displaying.

File: client/src/templates/Introduction/components/block.tsx (hunk 1 above)
Comment (moT01) [reply client/src/templates/Introduction/components/block.tsx:126]: We need to rework our challengeTypes - each type should have it's own type. The JS projects are the same type as the rest of the JS challenges, it should be its own.

**Comment by nhcarrigan** [comment]
@ojeytonwilliams realistically, would a better approach be a `isStepBased` key in the block meta? Which we could then use in the client to render it as a grid instead of a list?

**Comment by ojeytonwilliams** [comment]
@nhcarrigan yep, that sounds good. As long as the client just has to inspect a single property of a challenge, that's fine.

When we rework things, I think we'll be wanting to send the client block/superBlock info so it knows what to render. Our current approach of only passing challenge nodes and inferring everything from them is rather clunky, to say the least.

**Comment by nhcarrigan** [comment]
@JordanMooree our proposed solution is a bit larger in scope. Is this something you'd be interested in giving a shot?

**Comment by JordanMooree** [comment]
@nhcarrigan Yes, I'd be interested! Can you/anyone offer guidance on the proposed solution?

**Comment by nhcarrigan** [comment]
So the goal would be to add an `isStepBased: true` property to the `meta.json` files for the blocks that are step-based projects - then update that logic you were working on in `block.tsx` to look for that meta property instead of the logic we currently have.

There might be a couple of other places to update the logic too. If you'd like to give it a shot, you're welcome to push more commits to this branch. You can ping here when you're ready for another review, or ask in our [contributor chat](https://chat.freecodecamp.org/channel/contributors) if you have questions.

**Comment by JordanMooree** [comment]
Hey everyone, so I pushed a commit. Let me know what you think and any ideas on what I can do to further clean the code

**Comment by JordanMooree** [comment]
Just realized in my last commit I was adding `isStepBased` to blocks that were not in production yet. I fixed it and added the property to [Responsive Web Design](https://www.freecodecamp.org/learn/2022/responsive-web-design/) blocks.

**Comment by JordanMooree** [comment]
I will be closing this pull request and taking some time to understand Git and GitHub and reopen another pull request on this issue as soon as I can. I've been completing lost working on this through Git and GitHub 🙃

**Comment by raisedadead** [comment]
Hi, @JordanMooree no worries. There is no rush and we have all been there. If you need any help please find us in the Contributors chat room.

**Pull request closed without merging**